server/src/main/java/org/elasticsearch/action/support/replication/ReplicationOperation.java

```diff
@@ -187,9 +187,15 @@ public void onResponse(Void aVoid) {
             @Override
             public void onFailure(Exception e) {
                 logger.trace("[{}] op [{}] post replication actions failed for [{}]", primary.routingEntry().shardId(), opType, request);
-                // TODO: fail shard? This will otherwise have the local / global checkpoint info lagging, or possibly have replicas
-                // go out of sync with the primary
```

Comment:
I wonder about the last part of this todo? I see how this improves things and it seems ok to me. But should we leave the todo here for teh replicas go out of sync question? It is not entirely clear to me what is meant by that, do you know?

Reply:
I'm not entirely sure about what they meant with this TODO, my understanding is that we already fail the shard if the refresh fails or after a fsync fails. I'll leave the comment as is just in case.